NitroxModel/Serialization/ServerList.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using NitroxModel.Helper;
+using NitroxModel.Logger;
+
+namespace NitroxModel.Serialization
+{
+    public class ServerList
+    {
+        private const string SERVERS_FILE_NAME = "servers";
+        private static ServerList instance;
+        private readonly List<Entry> entries = new();
+        public static ServerList Instance => instance ??= From(DefaultFile);
+
+        private static ServerList Default
+        {
+            get
+            {
+                ServerList list = new();
+                list.Add(new Entry("local server", "127.0.0.1", 11000));
+                return list;
+            }
+        }
+
+        public static string DefaultFile => Path.Combine(NitroxAppData.Instance.LauncherPath, SERVERS_FILE_NAME);
+
+        public IEnumerable<Entry> Entries => entries;
+
+        public static ServerList From(string file = null)
+        {
+            // TODO: Remove backward compatibility after 1.5.0.0 release (no need to move old server file).
+            if (File.Exists(SERVERS_FILE_NAME))
+            {
+                File.Move(SERVERS_FILE_NAME, DefaultFile);
+            }
+
+            // Create file if it doesn't exist yet.
+            file ??= DefaultFile;
```

Comment:
While I am not against using this, I'm wondering if we should be using this until c# 8 becomes more widely known. Is there a possibility this might be a little unclear as it's not a widely known operator?

Reply:
It's a limited operator for null checks only. Similar to the limited features of `+=`, `/=` etc. If you know `??` you likely also know what `??=` does. I like using newer syntax because it keeps the code modern and concise and it's a good learning experience if you're unfamiliar.

Feedback like this really aught to come from someone actually having trouble understand the syntax so we can discuss the way forward. I think that, as long as the good is understood, that there is no issue.

Reply:
Yes I agree it's good to use new things, happy to continue use of it. Just wanted to throw a thought for people who may not have been familiar with the operator.